lib/web/desktop.go

```diff
@@ -354,8 +370,17 @@ func proxyWebsocketConn(ws *websocket.Conn, wds net.Conn) error {
 			if utils.IsOKNetworkError(err) {
 				errs <- nil
 				return
-			}
-			if err != nil {
+			} else if err != nil {
+				isFatal := !tdp.IsNonFatalErr(err)
```

Comment:
I wonder if we should refactor and make this `tdp.IsFatalErr`. For me it's much easier to think about than having to deal with the inverted form.

Reply:
I went with
```go
// IsNonFatalErr returns whether or not an error arising from
// the tdp package should be interpreted as fatal or non-fatal
// for an ongoing TDP connection.
func IsNonFatalErr(err error) bool {
	if err == nil {
		return false
	}

	return errors.Is(err, trace.LimitExceeded(clipDataMaxLenErrMsg)) ||
		errors.Is(err, trace.LimitExceeded(stringMaxLenErrMsg)) ||
		errors.Is(err, trace.LimitExceeded(fileReadWriteMaxLenErrMsg)) ||
		errors.Is(err, trace.LimitExceeded(mfaDataMaxLenErrMsg))
}

// IsFatalErr returns the inverse of IsNonFatalErr
// (except for if err == nil, for which both functions return false)
func IsFatalErr(err error) bool {
	if err == nil {
		return false
	}

	return !IsNonFatalErr(err)
}
```
and we can use whichever causes less double negatives in a particular context.